app/controllers/users_controller.rb

```diff
@@ -1,9 +1,20 @@
 # frozen_string_literal: true
 
 class UsersController < WithUserController
-  prepend_before_action { set_user(:id) }
-
   def show
     render json: @user
   end
+
+  def destroy
+    @user.user_counters.destroy_all
+    @user.favorite_tools.destroy_all
+    @user.destroy
```

Comment:
Don't you usually do `.destroy!` so as to throw errors? Most of the other controllers do.